datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -0,0 +1,83 @@
+package chaindatafetcher
+
+import (
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/log"
+	"github.com/klaytn/klaytn/networks/p2p"
+	"github.com/klaytn/klaytn/networks/rpc"
+	"github.com/klaytn/klaytn/node"
+	"sync"
+)
+
+var logger = log.NewModuleLogger(log.ChainDataFetcher)
+
+type ChainDataFetcher struct {
+	config *ChainDataFetcherConfig
+
+	scheduler Scheduler
+	handlers  []ChainDataHandler
+}
+
+func NewChainDataFetcher(ctx *node.ServiceContext, cfg *ChainDataFetcherConfig) (*ChainDataFetcher, error) {
+	reqCh := make(chan Request, cfg.RequestChSize)
+	resCh := make(chan Response, cfg.ResponseChSize)
+
+	s := NewScheduler(cfg.BlockChannelSize, reqCh, resCh)
+
+	var h []ChainDataHandler
+	for i := 0; i < cfg.NumHandlers; i++ {
+		h = append(h, NewChainDataHandler(i, reqCh, resCh))
+	}
+
+	f := &ChainDataFetcher{
+		config:    cfg,
+		scheduler: s,
+		handlers:  h,
+	}
+	return f, nil
+}
+
+func (f *ChainDataFetcher) Protocols() []p2p.Protocol {
+	return []p2p.Protocol{}
+}
+
+func (f *ChainDataFetcher) APIs() []rpc.API {
+	return []rpc.API{}
+}
+
+func (f *ChainDataFetcher) Start(server p2p.Server) error {
+	for _, h := range f.handlers {
+		go h.Start()
+	}
+	f.scheduler.Start()
+	logger.Info("chaindatafetcher is started")
+	return nil
+}
+
+func (f *ChainDataFetcher) Stop() error {
+	f.scheduler.Stop()
+	wg := sync.WaitGroup{}
+	for _, h := range f.handlers {
+		wg.Add(1)
+		go func(handler ChainDataHandler) {
```

Comment:
Do we need to use goroutines to stop the handlers?

Reply:
Nope. I modified logics to handle goroutines.